src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java

```diff
@@ -0,0 +1,189 @@
+import uk.gov.hmcts.reform.sscs.ccd.domain.PostHearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.PostponementRequest;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.docassembly.GenerateFile;
+import uk.gov.hmcts.reform.sscs.model.docassembly.GenerateFileParams;
+import uk.gov.hmcts.reform.sscs.model.docassembly.PdfRequestTemplateBody;
+
+@Slf4j
+public class PdfRequestUtil {
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static String requestDetails;
+    private static String title;
+    private static StringBuilder additionalRequestDetails;
+
+    private PdfRequestUtil() {
+        //
+    }
+
+    @AllArgsConstructor
+    private enum PdfType {
+        POSTPONEMENT("Postponement Request"),
+        POST_HEARING("Post Hearing Application");
+
+        final String name;
+
+        @Override
+        public String toString() {
+            return this.name;
+        }
+    }
+
+    public static PreSubmitCallbackResponse<SscsCaseData> processPostponementRequestPdfAndSetPreviewDocument(
+        String userAuthorisation,
+        SscsCaseData sscsCaseData,
+        PreSubmitCallbackResponse<SscsCaseData> response,
+        GenerateFile generateFile,
+        String templateId
+    ) {
+        return processRequestPdfAndSetPreviewDocument(PdfType.POSTPONEMENT, userAuthorisation, sscsCaseData, response, generateFile, templateId);
+    }
+
+    public static PreSubmitCallbackResponse<SscsCaseData> processPostHearingRequestPdfAndSetPreviewDocument(
+        String userAuthorisation,
+        SscsCaseData sscsCaseData,
+        PreSubmitCallbackResponse<SscsCaseData> response,
+        GenerateFile generateFile,
+        String templateId
+    ) {
+        return processRequestPdfAndSetPreviewDocument(PdfType.POST_HEARING, userAuthorisation, sscsCaseData, response, generateFile, templateId);
+    }
+
+    protected static PreSubmitCallbackResponse<SscsCaseData> processRequestPdfAndSetPreviewDocument(
+        PdfType pdfType,
+        String userAuthorisation,
+        SscsCaseData sscsCaseData,
+        PreSubmitCallbackResponse<SscsCaseData> response,
+        GenerateFile generateFile,
+        String templateId
+    ) {
+        log.debug("Executing processRequestPdfAndSetPreviewDocument for caseId: {}", sscsCaseData.getCcdCaseId());
+
+        additionalRequestDetails = new StringBuilder();
+        additionalRequestDetails.append("Date request received: ").append(LocalDate.now().format(DATE_TIME_FORMATTER)).append("\n");
+        StringBuilder pdfUrlBuilder = new StringBuilder();
+        switch (pdfType) {
+            case POST_HEARING:
+                String postHearingDocumentTypeLabel = getPostHearingDocumentType(sscsCaseData).getLabel();
+                pdfUrlBuilder.append(postHearingDocumentTypeLabel)
+                    .append(" from FTA"); // TODO SSCS-10759 make dynamic for the uploading party
```

Comment:
Note - Heads up, looks like changes for 10759 don't touch this code, as it's for scanned documents in ActionFurtherEvidence, but there is a possibility this will need to be updated in future, can leave TODO or update/remove it.

Reply:
removed